File: typescript/src/response.ts
import { PingStats } from "./stats.js";

export type PingResponse = {
  status: number;
};

export class ResponseHandler {
  updateStats(stats: PingStats, response: PingResponse) {
    if (response.status >= 200 && response.status < 300) {
      stats.addSuccess();
    } else {
      stats.addFail();
    }
  }
}

File: rust/src/response.rs
use crate::stats::PingStats;

pub struct PingResponse {
    pub status: u16,
}

pub struct ResponseHandler {}

impl ResponseHandler {
    pub fn update_stats(&self, stats: &mut PingStats, response: &PingResponse) {
        match response.status {
            200..=299 => stats.add_success(),
            _ => stats.add_fail(),
        };
    }
}
